server/pypi/packages/scipy/patches/chaquopy.patch

```diff
@@ -142,43 +195,53 @@ diff -aur src-orig/scipy/sparse/linalg/_propack/setup.py src/scipy/sparse/linalg
  
  def _is_32bit():
 -    return np.intp(0).itemsize < 8
--
 +    # Chaquopy: check 32/64 bit based on ABI instead of numpy
-+    # np.intp(0).itemsize < 8  
-+    return getenv('HOST') in ["arm-linux-androideabi", "i686-linux-android"]    
++    # return np.intp(0).itemsize < 8
++    return getenv('HOST') in ["arm-linux-androideabi", "i686-linux-android"]
+ 
  
  def check_propack_submodule():
-     if not (pathlib.Path(__file__).parent / 'PROPACK/README').exists():
-diff -aur src-orig/scipy/special/ellint_carlson_cpp_lite/ellint_arith_aux.hh src/scipy/special/ellint_carlson_cpp_lite/ellint_arith_aux.hh
---- src-orig/scipy/special/ellint_carlson_cpp_lite/ellint_arith_aux.hh	2023-09-20 11:08:03.696956933 +0200
-+++ src/scipy/special/ellint_carlson_cpp_lite/ellint_arith_aux.hh	2023-09-21 13:13:09.439909816 +0200
-@@ -5,6 +5,8 @@
- #include <cstddef>
- #include "ellint_typing.hh"
-
-+// Chaquopy: added <algorithm>
-+#include <algorithm>
- 
- /* Auxiliary floating-point manipulation utilities.
-  * Ref:
-diff -aur src-orig/scipy/special/_generate_pyx.py src/scipy/special/_generate_pyx.py
---- src-orig/scipy/special/_generate_pyx.py	2023-09-20 11:08:03.676956552 +0200
-+++ src/scipy/special/_generate_pyx.py	2023-09-21 13:13:09.443909897 +0200
-@@ -230,7 +230,7 @@
+diff --git a/scipy/spatial/_qhull.pyx b/scipy/spatial/_qhull.pyx
+index 8c9135adc..b6ff2d86f 100644
+--- a/scipy/spatial/_qhull.pyx
++++ b/scipy/spatial/_qhull.pyx
+@@ -17,6 +17,7 @@ cimport cython
+ from . cimport _qhull
+ from . cimport setlist
+ from libc cimport stdlib
++from libc.stdio cimport FILE # Chaquopy
+ from scipy._lib.messagestream cimport MessageStream
+ 
+ import os
+@@ -992,7 +993,7 @@ cdef class _Qhull:
+         return extremes_arr
+ 
+ 
+-cdef void _visit_voronoi(qhT *_qh, void *ptr, vertexT *vertex, vertexT *vertexA,
++cdef void _visit_voronoi(qhT *_qh, FILE *ptr, vertexT *vertex, vertexT *vertexA,
+                          setT *centers, boolT unbounded):
+     cdef _Qhull qh = <_Qhull>ptr
+     cdef int point_1, point_2, ix
+diff --git a/scipy/special/_generate_pyx.py b/scipy/special/_generate_pyx.py
+index 21b401629..15f84e553 100644
+--- a/scipy/special/_generate_pyx.py
++++ b/scipy/special/_generate_pyx.py
+@@ -232,8 +232,7 @@ import re
  import textwrap
  from typing import List
  
 -import numpy
+-
 +# Chaquopy: skip "import numpy"
```

Comment:
Was there some reason to change this? The previous version was less disruptive, as it retained the two lines after the imports.